### PR TITLE
update php versions to currently supported versions

### PR DESCRIPTION
**Pull request by cjschneider2**
update:
 - php versions to `7.4` & `8.0`
 - php unit to `^9` (also requires php 7.4 || 8.0)
fixes:
 - add return type for test interface requirement in `setup()`
tasks:
 - normalize `composer.json` with composer-normalize

### CODE REVIEW

**Comment by leon0399** [comment]
Hi! Thank you for your submission! I've just updated to native GitHub Actions, since appveyor not working. Could you, please, pull recent changes and update Actions config according to your changes?

**Comment by cjschneider2** [comment]
@leon0399 Sorry this update took so long, some other things got in the way :see_no_evil:

**Review by leon0399 — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -10,22 +12,20 @@
...
       "name": "Leonid Meleshin",
       "email": "leon.03.99@gmail.com"
     }
   ],
+  "homepage": "http://github.com/leon0399/monolog-flysytem",
   "require": {
-    "php": "^7.2",
-    "monolog/monolog": "^2.0",
-    "league/flysystem": "^1.0"
+    "php": ">=7.3",
```
Comment (leon0399) [inline composer.json:23]: Wouldn't it be better to use `>=7.4 || ^8`?

**Review by leon0399 — CHANGES_REQUESTED** [review]

**Comment by leon0399** [comment]
May I ask you to also modify the Actions file, removing 7.2 and adding 8.0, 8.1 and, maybe, 8.2 (since it is almost here 😄)

File: composer.json (hunk 1 above)
Comment (cjschneider2) [reply composer.json:23]: I'm pretty sure `>=7.4` also covers `^8` as well (in that `>=7.4` is equivalent to `^7.4 | ^8`). I just chose 7.3 as the minimum as it's the minimum that phpunit 9 supports. 

I can change it to which ever you prefer. I just personally need it compatible with the currently supported PHP version (8.1).

**Review by leon0399 — APPROVED** [review]
Excellent work! Thank you so much for your contribution!

**Pull request merged**